### PR TITLE
Reduce unnecessary call to computeLayout in renderTo()

### PR DESCRIPTION
**Pull request by hellochar**
register and then flush the component to renderTo it. Previously,
computeLayout and render were both invoked two times per renderTo().
This should have no regression in behavior but charts will render faster.

Addresses first point of #3117

### CODE REVIEW

**Review by CalvinFernandez — CHANGES_REQUESTED** [review]
Looks good! Agree with these changes for the most part.

1) This feels like render whack-a-mole. Did you investigate queueing renders and canceling multiple requested renders within the same render loop? Not necessarily asking for change here, just wondering what your thought process was.

2)  We can't remove exported functions without a major rev. Can you deprecate `registerToComputeLayout` and file an issue to remove in 3.x instead of totally removing it? 

3) Can we get at least 1 test proving this reduces the traffic?

**Comment by hellochar** [comment]
1. Requesting a render puts it in a Set of components-to-render-on-the-next-flush (which is usually in a requestAnimationFrame), so multiple renders are already a no-op since it's just adding to a queue. The whack-a-mole is in "requesting a render (this is calling `.render()`)" in the first place which could happen many times. This shouldn't matter because requesting a render should be a trivial operation, but Bar plots do computation as part of requesting a render in the first place. This is tracked in #2495 and I tried fixing this for a few hours but couldn't understand exactly what was happening so gave up on it for now.

2. Will do (it's just a rename anyways)

3. Yeah I'll write a test

**Comment by CalvinFernandez** [comment]
👍 cool just the rename and we're good

File: test/components/componentTests.ts [hunk 1]
```diff
@@ -735,6 +739,34 @@ describe("Component", () => {
...
+      const legend = new Plottable.Components.Legend(colorScale);
+      const table = new Plottable.Components.Table([[barPlot, legend]]);
+      const computeLayoutSpy = sinon.spy(table, "computeLayout");
+
+      table.renderTo(svg);
+
+      assert.strictEqual(computeLayoutSpy.callCount, 1, "component only computes layout once");
+      table.destroy();
+      svg.remove();
+      done();
```
Comment (CalvinFernandez) [inline test/components/componentTests.ts:767]: you only need `done` for async things

**Review by CalvinFernandez — APPROVED** [review]

**Pull request merged**